### PR TITLE
Add documentation for callNullFree

### PR DESCRIPTION
**Pull request by kevinwilfong**
Summary: Adding documentation for the callNullFree function to the documentation for Scalar/Simple functions as a sub-section under "Null Behavior".

Differential Revision: D34048567

### CODE REVIEW

File: velox/docs/develop/scalar-functions.rst [hunk 1]
```diff
@@ -104,6 +104,58 @@ an artificial example of a ceil function that returns 0 for null input:
...
+may be or contain null. Only if it can definitively be determined that that are
+no nulls will "callNullFree" be invoked.  In this case, "callNullFree" can act
+as a fast path by avoiding any per row null checks.
+
+Here is an example of an array_min function that returns the minimum value in
+an array:
+
+.. code-block:: c++
+
+				template <typename TExecParams>
```
Comment (laithsakka) [inline velox/docs/develop/scalar-functions.rst:127]: can this be written with proper indentation.

File: velox/docs/develop/scalar-functions.rst [hunk 2]
```diff
@@ -104,6 +104,58 @@ an artificial example of a ceil function that returns 0 for null input:
...
+				struct ArrayMinFunction {
```
Comment (laithsakka) [inline velox/docs/develop/scalar-functions.rst:128]: nit: consider a simpler example.

File: velox/docs/develop/scalar-functions.rst (hunk 1 above)
Comment (kevinwilfong) [reply velox/docs/develop/scalar-functions.rst:127]: Ya, I'm not sure why some of the other code blocks in this file are indented so far over

File: velox/docs/develop/scalar-functions.rst (hunk 2 above)
Comment (kevinwilfong) [reply velox/docs/develop/scalar-functions.rst:128]: removed some uninteresting code

**Pull request closed without merging**